### PR TITLE
Apply migrations from plugins during `ckan db upgrade`

### PR DESCRIPTION
**Pull request by smotornyuk**
By default, `db upgrade` applies migrations from all enabled plugins. 
To run only core migrations, add `--skip-plugins` flag. To run only plugins' migrations, add `--skip-core` flag. 
`db pending-migration` suggests using `db upgrade` when it executed for migration. I haven't removed it, because it works as an introspection tool that shows a number of unapplied migrations from enabled plugins.

### CODE REVIEW

**Review by pdelboca — APPROVED** [review]

File: ckan/cli/db.py [hunk 1]
```diff
@@ -58,13 +58,34 @@ def clean():
...
+):
     """Upgrade the database.
     """
-    _run_migrations(plugin, version)
-    click.secho(u'Upgrading DB: SUCCESS', fg=u'green', bold=True)
+    if not skip_core:
+        _run_migrations(plugin, version)
+
+    if not skip_plugins and not plugin:
+        _migrate_plugins(apply=True)
```
[the remark this answers is not in the document]
Comment (smotornyuk) [reply ckan/cli/db.py:76]: I'm sorry, but it's too opinionated suggestion. In the same way, someone can tell that it's hard to follow big number of private methods. You are suggesting introducing three private methods instead of one for calling two of them inside a third one. That's not ideal either. 

And I may be not the native speaker of English, but I'm sure that there are no double negations here. `not skip and not plugin` - that's two different negative assertions. `not skip` itself can hardly be callded double negation, because it's not the same as `not not_run`(which, of course, would be a double negation)

So your suggestion from my point of view doesn't make any good.
